frontend-project/src/models/connect.d.ts

```diff
@@ -0,0 +1,31 @@
+import { MenuDataItem } from '@ant-design/pro-layout';
+import { GlobalModelState } from './global';
+import { DefaultSettings as SettingModelState } from '../../config/defaultSettings';
+import { UserModelState } from './user';
+import { StateType } from './login';
+
+export { GlobalModelState, SettingModelState, UserModelState };
+
+export interface Loading {
+  global: boolean;
+  effects: { [key: string]: boolean | undefined };
+  models: {
+    global?: boolean;
+    menu?: boolean;
+    setting?: boolean;
+    user?: boolean;
+    login?: boolean;
+  };
+}
+
+export interface ConnectState {
+  global: GlobalModelState;
+  loading: Loading;
+  settings: SettingModelState;
+  user: UserModelState;
+  login: StateType;
+}
+
+export interface Route extends MenuDataItem {
+  routes?: Route[];
```

Comment:
To jest interfejs rekurencyjnie odwołujący się do samego siebie?

Reply:
Tak